frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -1324,113 +1324,69 @@ export const multitabEditorLogic = kea<multitabEditorLogicType>([
             },
         ],
     }),
-    urlToAction(({ actions, values, props }) => ({
+    urlToAction(({ actions, props }) => ({
         [urls.sqlEditor()]: async (_, searchParams) => {
             if (!searchParams.open_query && !searchParams.open_view && !searchParams.open_insight) {
                 return
             }
 
-            let tabAdded = false
-
-            const createQueryTab = async (): Promise<void> => {
-                if (searchParams.open_view) {
-                    // Open view
-                    const viewId = searchParams.open_view
-
-                    if (values.dataWarehouseSavedQueries.length === 0) {
-                        await dataWarehouseViewsLogic.asyncActions.loadDataWarehouseSavedQueries()
-                    }
-
-                    const view = values.dataWarehouseSavedQueries.find((n) => n.id === viewId)
-                    if (!view) {
-                        lemonToast.error('View not found')
-                        return
-                    }
-
-                    const queryToOpen = searchParams.open_query ? searchParams.open_query : view.query.query
+            // Wait for Monaco to be ready before proceeding
+            if (!props.monaco) {
+                return
+            }
 
-                    actions.editView(queryToOpen, view)
-                    tabAdded = true
-                    router.actions.replace(router.values.location.pathname)
-                } else if (searchParams.open_insight) {
+            try {
+                if (searchParams.open_insight) {
                     if (searchParams.open_insight === 'new') {
                         // Add new blank tab
                         actions.createTab()
-                        tabAdded = true
                         router.actions.replace(router.values.location.pathname)
                         return
                     }
 
-                    // Open Insight
+                    // Load insight data first
                     const shortId = searchParams.open_insight
                     const insight = await insightsApi.getByShortId(shortId, undefined, 'async')
                     if (!insight) {
                         lemonToast.error('Insight not found')
                         return
                     }
 
+                    // Get the query from the insight
                     let query = ''
                     if (insight.query?.kind === NodeKind.DataVisualizationNode) {
                         query = (insight.query as DataVisualizationNode).source.query
                     }
 
-                    const queryToOpen = searchParams.open_query ? searchParams.open_query : query
-
-                    actions.editInsight(queryToOpen, insight)
+                    // Create a new model with a unique URI to prevent state conflicts
+                    const uri = props.monaco.Uri.parse(`insight-${insight.short_id}-${Date.now()}`)
```

Comment:
it looks like this is the crux of the fix? Can this be worked into the existing logic rather than paving most of the previous logic?